subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/fingerprint/ConfigurationCacheFingerprintWriter.kt

```diff
@@ -641,13 +652,24 @@ class ConfigurationCacheFingerprintWriter(
         private
         val undeclaredEnvironmentVariables = newConcurrentHashSet<String>()
 
+        private
+        val remoteScriptsUris = newConcurrentHashSet<URI>()
+
         fun captureFile(file: File) {
             if (!capturedFiles.add(file)) {
                 return
             }
             write(inputFile(file))
         }
 
+        fun remoteScriptObserved(scriptSource: ScriptSource) {
+            val uri = scriptSource.resource.location.uri
+            if (uri != null && remoteScriptsUris.add(uri)) {
+                val contentHash = scriptSource.resource.contentHash
```

Comment:
❓ Could this access to `contentHash` here trigger a network roundtrip?

Reply:
Yes, it will

Reply:
I see. So it's indeed better to move the complex logic to the listener and let it honor `--offline`.

Reply:
Done